common/keygen.cpp

```diff
@@ -0,0 +1,98 @@
+#include <openssl/pem.h>
+#include <memory>
+#include <random>
+#include "keygen.h"
+
+namespace {
+
+	const int kKeySize = 4096;
+
+	std::string getPubKey(RSA* rsa) {
+		
+		std::unique_ptr<BIO, decltype(&BIO_free)> buf(BIO_new(BIO_s_mem()), BIO_free);
+		PEM_write_bio_RSAPublicKey(buf.get(), rsa);
+
+		size_t length = BIO_ctrl_pending(buf.get());
+
+		std::string res;
+		res.resize(length);
+
+		BIO_read(buf.get(), res.data(), length);
+
+		return res;
+	}
+
+	std::string getPrivKey(RSA* rsa) {
+		
+		std::unique_ptr<BIO, decltype(&BIO_free)> buf(BIO_new(BIO_s_mem()), BIO_free);
+		PEM_write_bio_RSAPrivateKey(buf.get(), rsa, nullptr, nullptr, 0, nullptr, nullptr);
+
+		size_t length = BIO_ctrl_pending(buf.get());
+		std::string res;
+		res.resize(length);
+
+		BIO_read(buf.get(), res.data(), length);
+
+		return res;
+	}
+
+}
+
+namespace andeme {
+
+	//PublicKey
+
+	PublicKey::PublicKey(std::string&& key) noexcept : public_key_{ std::move(key) } {}
```

Comment:
```suggestion
	PublicKey::PublicKey(std::string&& key) noexcept : public_key_(key) {}
```
У тебя есть конструктор принимающий универсальную ссылку
```c++
basic_string( basic_string&& other ) noexcept;
```

Reply:
std::move нужен, семантика перемещения не передаётся – имя параметра key является l-значением